src/test/java/org/example/framework/page/GooglePricingCalculatorEstimatePage.java

```
@@ -35,21 +35,8 @@ public List<String> getActualTextFromField() {
         return webElementTextList;
     }
 
-    public double getActualSumFromField() throws ParseException {
-        double parsedSum = 0d;
-        String elementText = getElementWithPresenceWait(WaitTimeouts.THREE_SEC, "//*[@id='compute']/descendant::b[contains(text(), 'Estimated Component Cost')]")
-                .getText();
-        Matcher matcher = Pattern.compile("([0-9,.]{2,20})").matcher(elementText);
-        if (matcher.find()) {
-            NumberFormat numberFormat = NumberFormat.getInstance(Locale.US);
-            parsedSum = numberFormat.parse(matcher.group()).doubleValue();
-        }
-        return parsedSum;
-    }
-
-    public double getEstimateSum() throws ParseException {
-        driver.switchTo().frame(0);
-        driver.switchTo().frame(iFrameElement);
+    @Step("Get sum from estimate field")
+    public double getSumFromEstimateField() throws ParseException {
```

What will happen if pattern not matched? It means no sum on page. You need to throw assertionError in that case
See rule 'fail fast'